src/Symfony/Component/HttpFoundation/Cookie.php

```diff
@@ -93,7 +93,7 @@ public function __construct(string $name, string $value = null, $expire = 0, ?st
         }
 
         // from PHP source code
-        if (preg_match("/[=,; \t\r\n\013\014]/", $name)) {
+        if ($raw && preg_match("/[=,; \t\r\n\013\014]/", $name)) {
```

Comment:
I think you got this test wrong. Now you only check the regex if `$raw` is `true` and ignore the regex if `$raw` is `false`. I guess it should be the other way around.

Just replace `&&` with `||` and it will be fine.

Reply:
I had this wrong in my original bug report as well (edited now). I'd expect this to be:
```
if (!$raw && preg_match("/[=,; \t\r\n\013\014]/", $name)) {
```
Changing it to || would always throw the exception in raw mode rather than skipping the check.

Reply:
This looks correct to me. The regex only needs checking in raw mode. In non-raw mode the output is later sanitised (urlencoded) in __toString so the check here isn't necessary.

It does look like the test suite also needs updating to only test for this exception when operating in raw mode though.

(Please ignore my previous comment on this, I was reading this backwards myself too!)

Reply:
ok, just ignore my comment too then.

Reply:
If I replace the condition with || the exception will be throw for any value if raw=true. 
The issue that we don't need to check the regex if cookie is not raw, because we encode the name and value anyway.